File: core/src/main/java/forpleuvoir/dhwu1a/core/event/message/GroupMessageEvent.java
package forpleuvoir.dhwu1a.core.event.message;

import forpleuvoir.dhwu1a.core.message.GroupMessage;
import forpleuvoir.dhwu1a.core.user.Group;
import forpleuvoir.dhwu1a.core.user.Member;

import javax.annotation.Nullable;

/**
 * 群消息事件
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.message
 * <p>#class_name GroupMessageEvent
 * <p>#create_time 2021/7/4 11:50
 */
public class GroupMessageEvent extends MessageEvent<GroupMessage> {
    /**
     * 事件相关群
     */
    public final Group group;
    /**
     * 事件相关群员
     */
    @Nullable
    public final Member member;

    public GroupMessageEvent(GroupMessage message) {
        super(message);
        this.group = message.group;
        this.member = message.getMember();
    }

    @Nullable
    public Member getMember() {
        return member;
    }

    @Override
    public Group getUser() {
        return group;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/message/GroupMessageEvent.kt
package forpleuvoir.dhwu1a.core.event.message

import forpleuvoir.dhwu1a.core.message.GroupMessage
import forpleuvoir.dhwu1a.core.user.Group
import forpleuvoir.dhwu1a.core.user.Member

/**
 * 群消息事件
 *
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.message
 *
 * #class_name GroupMessageEvent
 *
 * #create_time 2021/7/4 11:50
 */
class GroupMessageEvent(message: GroupMessage) : MessageEvent<GroupMessage?>(message) {
    /**
     * 事件相关群
     */
    val group: Group?

    /**
     * 事件相关群员
     */
    val member: Member?
    override fun getUser(): Group? {
        return group
    }

    init {
        group = message.group
        member = message.member
    }
}